util/Migrator/DbScripts/2020-11-18_00_EmergencyAccess.sql

```diff
@@ -0,0 +1,399 @@
+        AND (
+            (@OnlyUsers = 0 AND (EA.[Email] = @Email OR U.[Email] = @Email))
+            OR (@OnlyUsers = 1 AND U.[Email] = @Email)
+        )
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccess_ReadToNotify]
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        EA.*,
+        Grantee.Name as GranteeName,
+        Grantor.Email as GrantorEmail
+    FROM
+        [dbo].[EmergencyAccess] EA
+    LEFT JOIN
+        [dbo].[User] Grantor ON Grantor.[Id] = EA.[GrantorId]
+    LEFT JOIN
+        [dbo].[User] Grantee On Grantee.[Id] = EA.[GranteeId]
+    WHERE
+        EA.[Status] = 3
+    AND
+        DATEADD(DAY, EA.[WaitTimeDays] - 1, EA.[RecoveryInitiatedDate]) <= GETDATE()
+    AND
+        DATEADD(DAY, 1, EA.[LastNotificationDate]) <= GETDATE()
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccess_Update]
+    @Id UNIQUEIDENTIFIER,
+    @GrantorId UNIQUEIDENTIFIER,
+    @GranteeId UNIQUEIDENTIFIER,
+    @Email NVARCHAR(50),
+    @KeyEncrypted VARCHAR(MAX),
+    @Type TINYINT,
+    @Status TINYINT,
+    @WaitTimeDays SMALLINT,
+    @RecoveryInitiatedDate DATETIME2(7),
+    @LastNotificationDate DATETIME2(7),
+    @CreationDate DATETIME2(7),
+    @RevisionDate DATETIME2(7)
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    UPDATE
+        [dbo].[EmergencyAccess]
+    SET
+        [GrantorId] = @GrantorId,
+        [GranteeId] = @GranteeId,
+        [Email] = @Email,
+        [KeyEncrypted] = @KeyEncrypted,
+        [Type] = @Type,
+        [Status] = @Status,
+        [WaitTimeDays] = @WaitTimeDays,
+        [RecoveryInitiatedDate] = @RecoveryInitiatedDate,
+        [LastNotificationDate] = @LastNotificationDate,
+        [CreationDate] = @CreationDate,
+        [RevisionDate] = @RevisionDate
+    WHERE
+        [Id] = @Id
+
+    EXEC [dbo].[User_BumpAccountRevisionDate] @GranteeId
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccessDetails_ReadByGranteeId]
+    @GranteeId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        *
+    FROM
+        [dbo].[EmergencyAccessDetailsView]
+    WHERE
+        [GranteeId] = @GranteeId
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccessDetails_ReadByGrantorId]
+    @GrantorId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        *
+    FROM
+        [dbo].[EmergencyAccessDetailsView]
+    WHERE
+        [GrantorId] = @GrantorId
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccessDetails_ReadByIdGrantorId]
+    @Id UNIQUEIDENTIFIER,
+    @GrantorId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        *
+    FROM
+        [dbo].[EmergencyAccessDetailsView]
+    WHERE
+        [Id] = @Id
+    AND
+        [GrantorId] = @GrantorId
+END
+GO
+
+CREATE PROCEDURE [dbo].[EmergencyAccessDetails_ReadExpiredRecoveries]
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        *
+    FROM
+        [dbo].[EmergencyAccessDetailsView]
+    WHERE
+        [Status] = 3
+    AND
+        DATEADD(DAY, [WaitTimeDays], [RecoveryInitiatedDate]) <= GETDATE()
```

Comment:
Don't forget to update the `GETDATE()` instances in the migration script with the `GETUTCDATE()`